src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infrastructure/ckan/persistence/CkanFacetsQueryBuilder.java

```diff
@@ -20,41 +23,151 @@
+                .map(value -> QUOTED_VALUE.formatted(value))
                 .collect(joining(operator));
 
-        return FACET_PATTERN.formatted(key, values);
+        if (values.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(FACET_PATTERN.formatted(key, values));
+    }
+
+    private Optional<String> buildDateTimeRangeQuery(String key,
+            List<DatasetSearchQueryFacet> facets) {
+        return resolveDateTimeCondition(facets)
+                .flatMap(condition -> condition.toSolrQuery(key));
+    }
+
+    private Optional<DateTimeCondition> resolveDateTimeCondition(
+            List<DatasetSearchQueryFacet> facets) {
+        var dateTimeFacets = facets.stream()
+                .filter(facet -> FilterType.DATETIME.equals(facet.getType()))
+                .toList();
+
+        if (dateTimeFacets.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var lower = findValueByOperator(dateTimeFacets,
+                Operator.GREATER_THAN_OR_EQUAL_TO_SYMBOL,
+                Operator.GREATER_THAN_SYMBOL).orElse(null);
+
+        var upper = findValueByOperator(dateTimeFacets,
+                Operator.LESS_THAN_OR_EQUAL_TO_SYMBOL,
+                Operator.LESS_THAN_SYMBOL).orElse(null);
+
+        var exact = findValueByOperator(dateTimeFacets, Operator.EQUAL_SYMBOL).orElse(null);
+
+        if (lower == null && upper == null && exact == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(new DateTimeCondition(lower, upper, exact));
+    }
+
+    private Optional<String> findValueByOperator(
+            List<DatasetSearchQueryFacet> facets, Operator... operators
+    ) {
+        if (operators == null || operators.length == 0) {
+            return Optional.empty();
+        }
+
+        var allowedOperators = java.util.EnumSet.noneOf(Operator.class);
+        allowedOperators.addAll(List.of(operators));
+
+        return facets.stream()
+                .filter(facet -> facet.getOperator() != null &&
+                        allowedOperators.contains(facet.getOperator()))
+                .map(DatasetSearchQueryFacet::getValue)
+                .filter(value -> nonNull(value) && !value.isBlank())
+                .map(String::trim)
+                .findFirst();
+    }
+
+    private static String formatRangeBoundary(String value) {
+        return value == null || value.isBlank()
+                ? RANGE_WILDCARD
+                : QUOTED_VALUE.formatted(value);
+    }
+
+    private static final class DateTimeCondition {
+
+        private final String lowerBound;
+        private final String upperBound;
+        private final String exactMatch;
+
+        private DateTimeCondition(String lowerBound, String upperBound, String exactMatch) {
+            this.lowerBound = lowerBound;
+            this.upperBound = upperBound;
+            this.exactMatch = exactMatch;
+        }
```

Comment:
This could be a `record`. If it stays a `class`, there's no need for it to be `final`.